controlpanel/frontend/views/app.py

```diff
@@ -200,12 +200,28 @@ def get_success_url(self):
 
     def form_valid(self, form):
         try:
+            assume_role = form.cleaned_data.get("allow_cloud_platform_assume_role")
+            role_arn = form.cleaned_data.get("cloud_platform_role_arn")
+            if assume_role and not role_arn:
+                form.add_error("cloud_platform_role_arn", "Role ARN is required")
+                return FormMixin.form_invalid(self, form)
```

Comment:
Django forms have a `clean` method where you can add validation on fields that depend on each other - so you might want to have a look at moving this there to keep the logic out of the views. Link to docs https://docs.djangoproject.com/en/5.1/ref/forms/validation/#cleaning-and-validating-fields-that-depend-on-each-other